exporter/management/commands/flattener.py

```diff
@@ -32,37 +32,47 @@ def handle(self, *args, **options):
 
 def callback(state, channel, method, properties, input_message):
     job_id = input_message.get("job_id")
-    max_rows_lower_bound = Job.objects.get(id=job_id).get_max_rows_lower_bound()
-
+    file_path = input_message.get("file_path")
     export = Export(job_id, export_type="flat")
-    export.lock()
 
     # Acknowledge now to avoid connection losses. The rest can run for hours and is irreversible anyhow.
     ack(state, channel, method.delivery_tag)
 
-    for entry in os.scandir(export.directory):
-        if not entry.name.endswith(".jsonl.gz") or "_" in entry.name:  # don't process months at the moment
-            continue
+    if file_path:
+        process_file(file_path, job_id)
+    else:
+        for entry in os.scandir(export.directory):
+            if not entry.name.endswith(".jsonl.gz") or "_" in entry.name:  # don't process months at the moment
+                continue
+            publish({"job_id": job_id, "file_path": entry.path}, "flattener_init")
```

Comment:
The `consume()` call above should be changed to `consume(callback, "flattener_init", ["flattener_init", "flattener_file"], decorator=decorator)` and then the `publish()` call can use `"flattener_file"`.

That way, different message formats have different routing keys, and we can decide to use a separate worker with a separate queue for the file-processing in the future.

Reply:
👍🏻 I thought about creating a new worker already or not, but I'm not sure if we want one worker just for creating others. As you said, we can decide in the future.

Reply:
In Kingfisher Process, file_worker just creates a few DB records and publishes messages. But in this case we really are not doing anything except creating messages, so I think one worker is fine.